src/rev_ai/streamingclient.py

```diff
@@ -62,19 +62,24 @@ def __init__(self,
         self.on_connected = on_connected
         self.client = websocket.WebSocket(enable_multithread=True)
 
-    def start(self, generator, metadata=None):
+    def start(self, generator, metadata=None, custom_vocabulary_id=None):
         """Function to connect the websocket to the URL and start the response
             thread
 
         :param generator: generator object that yields binary audio data
         :param metadata: metadata to be attached to streaming job
         """
-        url = self.base_url + '?access_token={}'.format(self.access_token) \
-            + '&content_type={}'.format(self.config.get_content_type_string()) \
-            + '&user_agent={}'.format(quote('RevAi-PythonSDK/{}'.format(__version__), safe=''))
+        url = self.base_url + '?' + urlencode({
```

Comment:
it would have made more sense to minimize string manipulation and prepare the query parameters first, then only serialize once.

e.g.
```
params = {
...
}
if cond:
  params.update(...)
if cond2:
  params.update(...)

url = self.base_url + '?' + urlencode(params)
```